docs/designers-developers/developers/block-api/block-context.md

```diff
@@ -0,0 +1,72 @@
+# Block Context
+
+Block context is a feature which enables ancestor blocks to provide values which can be consumed by descendent blocks within its own hierarchy. Those descendent blocks can inherit these values without resorting to hard-coded values and without an explicit awareness of the block which provides those values.
+
+This is especially useful in full-site editing where, for example, the contents of a block may depend on the context of the post in which it is displayed. A blogroll template may show excerpts of many different posts. Using block context, there can still be one single "Post Excerpt" block which displays the contents of the post based on an inherited post ID.
+
+If you are familiar with [React Context](https://reactjs.org/docs/context.html), block context adopts many of the same ideas. In fact, the client-side block editor implementation of block context is a very simple application of React Context. Block context is also supported in server-side `render_callback` implementations, demonstrated in the examples below.
+
+## Defining Block Context
+
+Block context is defined in the registered settings of a block. A block can provide a context value, or consume a value it seeks to inherit.
+
+### Providing Block Context
+
+A block can provide a context value by assigning a `providesContext` property in its registered settings. This is an object which maps a context name to one of the block's own attribute. The value corresponding to that attribute value is made available to descendent blocks and can be referenced by the same context name. Currently, block context only supports values derived from the block's own attributes. This could be enhanced in the future to support additional sources of context values.
+
+`record/block.json`
+
+```json
+{
+	"name": "my-plugin/record",
+	"attributes": {
+		"recordId": {
+			"type": "number"
+		}
+	},
+	"providesContext": {
+		"my-plugin/recordId": "recordId"
+	}
```

Comment:
Maybe foreshadowing a future point of possible confusion: The actual implementation of this was accidentally reversed in the browser implementation, treating the keys as the _attribute names_.

https://github.com/WordPress/gutenberg/blob/305844bb15e70a134174a65a7bd8dbea91c30f7c/packages/block-editor/src/components/inner-blocks/index.js#L58-L61

It's a simple enough fix (use `mapValues` to assign context value by attribute name), but raising it anyways, since if I was confused in the implementation, there's a chance developers may encounter similar confusion. It's documented pretty clearly here, so I don't know that there's much more we can do.

Reply:
>The actual implementation of this was accidentally reversed in the browser implementation, treating the keys as the _attribute names_.

Fixed in 1d44e76.